src/operator/concat-static.ts

```diff
@@ -12,12 +12,12 @@ import {isScheduler} from '../util/isScheduler';
  * @params {Scheduler} [scheduler] an optional scheduler to schedule each observable subscription on.
  * @returns {Observable} All values of each passed observable merged into a single observable, in order, in serial fashion.
  */
-export function concat<R>(...observables: Array<Observable<any> | Scheduler>): Observable<R> {
+export function concat<T, R>(...observables: Array<Observable<any> | Scheduler>): Observable<T> {
```

Comment:
What'll happen if 2 source has different types like

```
var source1 = Rx.Observable.of(42);
var source2 = Rx.Observable.of('b');

var source = Rx.Observable.concat(source1, source2);
```

and try to subscribe into `source`?